File: python/selection.py
import util

# sort implements selection sort algorithm. The function takes an array.
def sort(a):
	for i in range(len(a)):
		k = i
		for j in range(i+1, len(a)):
			if a[j] < a[k]:
				k = j
		util.swap(a, i, k)

File: java/Selection.java
public class Selection {
	public static void sort(int a[]) {
		for(int i = 0; i < a.length; i++) {
			int k = i;
			for(int j = i+1; j < a.length; j++) {
				if(a[j] < a[k])
					k = j;
			}
			Util.swap(a, i, k);
		}
	}
}
